.github/workflows/java-gradle-artifact.yaml

```diff
@@ -0,0 +1,171 @@
+        required: true
+      github-token:
+        description: "GitHub token for requesting changes from API."
+        required: true
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Build
+        uses: bakdata/ci-templates/actions/java-gradle-build@feature/java-gradle
+        with:
+          java-distribution: ${{ inputs.java-distribution }}
+          java-version: ${{ inputs.java-version }}
+          gradle-version: ${{ inputs.gradle-version }}
+          gradle-cache: ${{ inputs.gradle-cache }}
+          working-directory: ${{ inputs.working-directory }}
+
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
+    needs: build
+
+    steps:
+      - name: Test
+        uses: bakdata/ci-templates/actions/java-gradle-test@feature/java-gradle
+        with:
+          integration-tests: true
+          sonar-token: ${{ secrets.sonar-token }}
+          signing-secret-key-ring: ${{ secrets.signing-secret-key-ring }}
+          signing-key-id: ${{ secrets.signing-key-id }}
+          signing-password: ${{ secrets.signing-password }}   
+          java-distribution: ${{ inputs.java-distribution }}
+          java-version: ${{ inputs.java-version }}
+          gradle-version: ${{ inputs.gradle-version }}
+          gradle-cache: ${{ inputs.gradle-cache }}       
+          working-directory: ${{ inputs.working-directory }}
+
+  build-jib:
+    name: Build Jib Image
+    runs-on: ubuntu-22.04
+    needs: test
+
+    steps:
+      - name: Build Jib Image
+        uses: bakdata/ci-templates/actions/java-gradle-build-jib-image@feature/java-gradle
+        with:
+          java-distribution: ${{ inputs.java-distribution }}
+          java-version: ${{ inputs.java-version }}
+          gradle-version: ${{ inputs.gradle-version }}
+          gradle-cache: ${{ inputs.gradle-cache }}
+          working-directory: ${{ inputs.working-directory }}
+
+  publish:
+    name: Publish
+    runs-on: ubuntu-22.04
+    needs: build-jib
+
+    steps:
+      - name: Publish
+        uses: bakdata/ci-templates/actions/java-gradle-publish@feature/java-gradle
+        with:
+          signing-secret-key-ring: ${{ secrets.signing-secret-key-ring }}
+          signing-key-id: ${{ secrets.signing-key-id }}
+          signing-password: ${{ secrets.signing-password }}
+          ossrh-username: ${{ secrets.ossrh-username }}
+          ossrh-password: ${{ secrets.ossrh-password }}
+          java-distribution: ${{ inputs.java-distribution }}
+          java-version: ${{ inputs.java-version }}
+          gradle-version: ${{ inputs.gradle-version }}
+          gradle-cache: ${{ inputs.gradle-cache }}
+          working-directory: ${{ inputs.working-directory }}
+
+  publish-jib-image:
+    name: Publish Jib Image
+    runs-on: ubuntu-22.04
+    needs: build-jib
+
+    steps:
+    - name: Publish jib image
+      uses: bakdata/ci-templates/actions/docker-publish@feature/java-gradle
```

Comment:
what about proper tagging with latest and using release tags?

Reply:
Tagging is done in the action, see https://github.com/bakdata/ci-templates/pull/24/files/2fe7cd87d2564b00575035cad627b20d0d5858a8#diff-49358adb4ef9194dc03c4e9aa242a9637587a9707e7eb4e0d48d78e096b86369R50-R58